cmd/runtimetest/main.go

```diff
@@ -370,6 +371,52 @@ func validateMountsExist(spec *rspec.Spec) error {
 	return nil
 }
 
+func getActualCgroupPath() (string, error) {
+	f, err := os.Open("/proc/self/cgroup")
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if err := s.Err(); err != nil {
+			return "", err
+		}
+
+		text := s.Text()
+		parts := strings.Split(text, ":")
+		return parts[2], nil
```

Comment:
On Sun, Jul 31, 2016 at 09:36:26PM -0700, W. Trevor King wrote:
> > For all controllers needed for `linux.resources` (may be an empty
> > set), ensure that they have the same path.  Return that path for
> > `validateCgroupsPath`.

This would miss runtimes that for whatever reason attached an unneeded
controller with an inconsistent path.  Better logic:

* Pass the host controller paths in to the container (like we already
  pass config.json in).  This could be via a new file, or via
  config.json's annotations.
* For each entry in /proc/self/cgroup, assert that the path is the
  same as the host path for that controller, or some new path.  If
  there are multiple controllers with new paths, ensure they all
  match.

In pseudocode:

  var containerPath *string
  for s.Scan() {
    …
    parts := strings.Split(text, ":")
    // should probably check to make sure len(parts) == 3
    hierarchyId := parts[0]
    newPath := parts[2]
    oldPath := oldPaths[hierarchyId]
    if newPath == oldPath {
      controllers := strings.Split(parts[1], ",")
      for _, controller := range(controllers) {
        err = assertCgroupUnchanged(controller, oldControllerConfigs[controller])
        // log the error
      }
    } else if containerPath == nil {
      containerPath = newPath
    } else if newPath != containerPath {
      // log the error
    }
  }
  return containerPath, nil